cg/services/illumina/file_parsing/models.py

```diff
@@ -28,3 +29,41 @@ class DemuxMetrics(BaseModel):
     lane: int = Field(..., alias=DemuxMetricsColumnNames.LANE)
     sample_internal_id: str = Field(..., alias=DemuxMetricsColumnNames.SAMPLE_INTERNAL_ID)
     read_pair_count: int = Field(..., alias=DemuxMetricsColumnNames.READ_PAIR_COUNT)
+
+
+class DsmcEncryptionKey(BaseModel):
+    """Model representing the response from a PDC query."""
+
+    date: str
+    key_path: str
+
+    @field_validator("date")
+    def parse_date(cls, value: str) -> datetime:
+        return datetime.strptime(value, "%m/%d/%Y %H:%M:%S")
+
+    @field_validator("key_path")
+    def validate_sequencing_path(cls, value: str) -> str:
+        if not value.endswith(f"{FileExtensions.KEY}{FileExtensions.GPG}"):
+            raise ValueError(f'"{value}" - is not the path to the Encryption key')
+        if not Path(value):
+            raise ValueError(f'"{value}" - is not a valid file path.')
+        return value
```

Comment:
I would prefer if this validator returned a `Path` object. Then we could write this as:
```suggestion
    def validate_sequencing_path(cls, value: str) -> str:
        if not value.endswith(f"{FileExtensions.KEY}{FileExtensions.GPG}"):
            raise ValueError(f'"{value}" - is not the path to the Encryption key')
        return Path(value)
```

I am a bit unsure of what the `if not Path(value)` tests as well. Is it only if the "value" would be set to "None"?